pkg/controllers/cloud/account_poller.go

```diff
@@ -243,20 +243,20 @@ func (p *accountPoller) doAccountPoller() {
 		updateAccountStatus(&account.Status, &discoveredStatus)
 		e = p.Client.Status().Update(context.TODO(), account)
 		if e != nil {
-			p.log.Info("failed to update account status", "account", p.namespacedName, "err", e)
+			p.log.Error(e, "failed to update account status", "account", p.namespacedName)
 		}
 	}
 
 	vpcMap, e := cloudInterface.GetVpcInventory(p.namespacedName)
 	vpcCount := len(vpcMap)
 	if e != nil {
-		p.log.Info("failed to fetch cloud vpc list from internal snapshot", "account",
-			p.namespacedName.String(), "error", e)
+		p.log.Error(e, "failed to fetch cloud vpc list from internal snapshot", "account",
+			p.namespacedName.String())
 	}
 
-	err := p.inventory.BuildVpcCache(vpcMap, p.namespacedName)
-	if err != nil {
-		p.log.Info("failed to build vpc inventory", "account", p.namespacedName.String(), "error", err)
+	e = p.inventory.BuildVpcCache(vpcMap, p.namespacedName)
+	if e != nil {
+		p.log.Error(e, "failed to build vpc cache", "account", p.namespacedName.String())
```

Comment:
logic point of view, if there is error is building vpc cache, is there any point continuing further?

Reply:
p.inventory.BuildVpcCache - returns an error only when creating or updating a VPC..
When it fails, we should still continue to pull VM inventory, because VM resources might have changed.


vpcMap, e := cloudInterface.GetVpcInventory(p.namespacedName) - returns an error when configs are not present in cache. It implies CPA config is invalid. We should not continue forward and return when an error occurs.